tests/unittests/test_request_timeout.py

```diff
@@ -0,0 +1,143 @@
+import logging
+import unittest
+from unittest import mock
+import tap_snapchat_ads
+
+
+class MockParseArgs:
+    """Mock the parsed_args() in main"""
+
+    config = {}
+    state = {}
+    catalog = {}
+    discover = False
+
+    def __init__(self, config, state, catalog, discover):
+        self.config = config
+        self.state = state
+        self.catalog = catalog
+        self.discover = discover
+
+
+def get_args(config, state, catalog, discover):
+    """Return the MockParseArgs object"""
+    return MockParseArgs(config, state, catalog, discover)
+
+
+@mock.patch("tap_snapchat_ads.SnapchatClient")
+@mock.patch("singer.utils.parse_args")
+class TestTimeoutValue(unittest.TestCase):
+    """
+    Test that "request_timeout" parameter works properly in various cases
+    """
+
+    mock_config = {
+        "client_id": "YOUR_CLIENT_ID",
+        "client_secret": "YOUR_CLIENT_SECRET",
+        "refresh_token": "YOUR_REFRESH_TOKEN",
+        "user_agent": "tap-snapchat-ads api_user_email@example.com",
+    }
+
+    def test_no_timeout_value_in_config(self, mocked_parse_args, mocked_client):
+        """
+        Unit test to ensure that default "request_timeout" value is used when "requst_timeout"  is not passed in config
+        """
+        mock_config = {
+            "client_id": "YOUR_CLIENT_ID",
+            "client_secret": "YOUR_CLIENT_SECRET",
+            "refresh_token": "YOUR_REFRESH_TOKEN",
+            "user_agent": "tap-snapchat-ads api_user_email@example.com",
+        }
```

Comment:
Remove this from here and use already defined `mock_config` at the class level.

Reply:
class variable mock_config is modified by another test function. So, we need to explicitly pass this config in this function.